jib-core/src/main/java/com/google/cloud/tools/jib/docker/DockerClient.java

```diff
@@ -111,51 +85,26 @@ public static DockerClient newDefaultClient() {
     };
   }
 
-  private static final Path DEFAULT_DOCKER_CLIENT = Paths.get("docker");
+  public static final Path DEFAULT_DOCKER_CLIENT = Paths.get("docker");
 
   /** Factory for generating the {@link ProcessBuilder} for running {@code docker} commands. */
   private final Function<List<String>, ProcessBuilder> processBuilderFactory;
 
-  @VisibleForTesting
-  DockerClient(Function<List<String>, ProcessBuilder> processBuilderFactory) {
-    this.processBuilderFactory = processBuilderFactory;
-  }
-
   /**
    * Instantiates with a {@code docker} executable and environment variables.
    *
-   * @param dockerExecutable path to {@code docker}
    * @param dockerEnvironment environment variables for {@code docker}
+   * @param dockerExecutable path to {@code docker}
    */
-  private DockerClient(Path dockerExecutable, ImmutableMap<String, String> dockerEnvironment) {
-    this(defaultProcessBuilderFactory(dockerExecutable.toString(), dockerEnvironment));
-  }
-
-  /**
-   * Checks if Docker is installed on the user's system and accessible by running the default {@code
-   * docker} command.
-   *
-   * @return {@code true} if Docker is installed on the user's system and accessible
-   */
-  public static boolean isDefaultDockerInstalled() {
-    return isDockerInstalled(DEFAULT_DOCKER_CLIENT);
+  public DockerClient(Map<String, String> dockerEnvironment, Path dockerExecutable) {
```

Comment:
Perhaps switch the order of the arguments. `defaultProcessBuilderFactory` takes them in the opposite order. I think `dockerExecutable` path is more important to be first.

Reply:
Oh yeah, I'd switched them because I made one of them nullable, but then changed my mind and forgot to switch them back.